pkg/integrations/downloader/downloader.go

```diff
@@ -0,0 +1,137 @@
+// Package downloader implements a file downloader that can download files from
+// a URL and verify checksums of the file.
+package downloader
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"sync"
+)
+
+// Global is a globally shared downloader.
+var Global = New(nil)
```

Comment:
This isn't necessary for this PR, but we'll need it eventually: if we're going to have multiple instances of `jmx_exporter` running, we'll need to handle the case where multiple instances might be trying to download the same file concurrently, which this code is able to manage.